plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/DBUtils.java

```diff
@@ -1278,18 +1278,22 @@ public static DBCStatement makeStatement(
         long maxRows)
         throws DBCException
     {
+        DBCQueryTransformProvider transformProvider = DBUtils.getAdapter(DBCQueryTransformProvider.class, session.getDataSource());
+        
         // We need to detect whether it is a plain select statement
         // or some DML. For DML statements we mustn't set limits
         // because it sets update rows limit [SQL Server]
         boolean selectQuery = sqlQuery.getType() == SQLQueryType.SELECT && sqlQuery.isPlainSelect();
+        // For some DB's like Vertica LIMIT can't be applied to queries without FROM part - i.e calling meta functions.
+        if (sqlQuery.isPlainSelectWithoutFrom() && transformProvider.getClass().getSimpleName().equals("VerticaDataSource"))
```

Comment:
I can suggest you to extend [SQLDialect.java](https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/sql/SQLDialect.java) and add convenient method to determine whether the `LIMIT` without `FROM` is allowed or not, e.g:
```java
boolean supportsLimitWithoutFrom();
```
And then override it to return `true` exclusively for [VerticaSQLDialect.java](https://github.com/dbeaver/dbeaver/blob/devel/plugins/org.jkiss.dbeaver.ext.vertica/src/org/jkiss/dbeaver/ext/vertica/model/VerticaSQLDialect.java). It would be more generic approach.

Reply:
You can obtain dialect from the data source (which is obtained from the available `session` variable in this case):
```java
final SQLDialect dialect = SQLUtils.getDialectFromObject(session.getDataSource());
```

Reply:
I think I found it using slightly other access:

session.getDataSource().getSQLDialect()

Is it good too?

Reply:
That's okay too :)